testsuite/podman_runner/03_run_controller_and_registry_and_buildhost.sh

```diff
@@ -2,28 +2,150 @@
+$PODMAN_CMD run -d --network network --name ${NO_AUTH_REGISTRY_NAME} -h ${NO_AUTH_REGISTRY_NAME} -e NO_AUTH_REGISTRY=${NO_AUTH_REGISTRY_NAME} -e REGISTRY_HTTP_ADDR="0.0.0.0:5000" -p 5002:5000 docker.io/library/registry:2
+
+$PODMAN_CMD run --privileged -d --network network -v ${src_dir}/testsuite:/testsuite -v /tmp/testing/buildhost_product_uuid:/sys/class/dmi/id/product_uuid -v /tmp/testing:/tmp -v ${src_dir}/testsuite/podman_runner/salt-minion-entry-point.sh:/salt-minion-entry-point.sh --volume /run/dbus/system_bus_socket:/run/dbus/system_bus_socket:ro -v /var/run/docker.sock:/var/run/docker.sock --name buildhost -h buildhost ghcr.io/$UYUNI_PROJECT/uyuni/ci-buildhost:$UYUNI_VERSION bash -c "/salt-minion-entry-point.sh server 1-SUSE-KEY-x86_64"
+$PODMAN_CMD exec -d buildhost dockerd
+$PODMAN_CMD exec buildhost bash -c "sed -e 's/http:\/\/download.opensuse.org/file:\/\/\/mirror\/download.opensuse.org/g' -i /etc/zypp/repos.d/*"
+$PODMAN_CMD exec buildhost bash -c "sed -e 's/https:\/\/download.opensuse.org/file:\/\/\/mirror\/download.opensuse.org/g' -i /etc/zypp/repos.d/*"
+
+$PODMAN_CMD ps
+
+# -----------------------------------------------------------------------
+## 1. WAIT FOR REGISTRIES TO BE UP AND LISTENING
+# -----------------------------------------------------------------------
+
+REGISTRIES_TO_CHECK=("${AUTH_REGISTRY_HOST_ADDR}" "${NO_AUTH_REGISTRY_HOST_ADDR}")
+TIMEOUT=30
+for REGISTRY_ADDR in "${REGISTRIES_TO_CHECK[@]}"; do
+    echo "Waiting for registry ${REGISTRY_ADDR} to respond..."
+    start_time=$(date +%s)
+    # Check for 200/401 response over HTTP (using the same protocol the push will use)
+    while ! curl -s -o /dev/null -w "%{http_code}" "http://${REGISTRY_ADDR}/v2/" | grep -E '200|401' &>/dev/null; do
+        current_time=$(date +%s)
+        elapsed=$((current_time - start_time))
+        if [ "$elapsed" -ge "$TIMEOUT" ]; then
+            echo "Error: Registry ${REGISTRY_ADDR} failed to start after ${TIMEOUT} seconds."
+            exit 1
+        fi
+        sleep 1
+    done
+    echo "Registry ${REGISTRY_ADDR} is up."
+done
+
+# -----------------------------------------------------------------------
+## 2. CONFIGURE HOST/VM PODMAN FOR INSECURE REGISTRIES (V2 Format)
+# -----------------------------------------------------------------------
+
+if [[ "$(uname)" == "Darwin" ]]; then
+    HOST_REGISTRIES_CONF_NAME="99-insecure-lab-registry.conf"
+    HOST_REGISTRIES_CONF_PATH="/tmp/${HOST_REGISTRIES_CONF_NAME}"
+    TARGET_DIR="/etc/containers/registries.conf.d"
+
+    echo "Configuring Podman environment for insecure registry ${NO_AUTH_REGISTRY_HOST_ADDR} in V2 format..."
+
+    CONFIG_CONTENT=$(cat <<EOF
+[[registry]]
+location = "${NO_AUTH_REGISTRY_HOST_ADDR}"
+insecure = true
+EOF
+)
+
+    echo "${CONFIG_CONTENT}" > ${HOST_REGISTRIES_CONF_PATH}
```

Comment:
```suggestion
    cat > ${HOST_REGISTRIES_CONF_PATH} <<EOF
[[registry]]
location = "${NO_AUTH_REGISTRY_HOST_ADDR}"
insecure = true
EOF 
```

Should work too and be more concise